core/src/main/java/org/apache/iceberg/BaseContentScanTask.java

```diff
@@ -117,4 +121,19 @@ public String toString() {
         .add("residual", residual())
         .toString();
   }
+
+  public static long estimatedRowsCount(long length, ContentFile<?> file) {
```

Comment:
It is the same logic as in `ContentScanTask` but expressed using arrays. Exposed here to be reused in split tasks.

Reply:
Yeah, we could do that. I renamed it to `estimateRowsCount` as it actually does the estimation.